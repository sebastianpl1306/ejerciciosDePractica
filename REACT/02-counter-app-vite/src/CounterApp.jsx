import { useState } from 'react';
import PropTypes from "prop-types";

export const CounterApp =  ({ value }) => {

    //El useState recibe como parametro el valor de inicialización y este se desestructura en una variable llamada counter y una función que permite modificar
    //el valor de counter
    const [ counter, setCounter ] = useState( value );

    const handleAdd = () =>{
        // setCounter( counter + 1 );
        // setCounter(1000);
        setCounter((c)=> c + 1);
    }

    const handleSubstract = () => setCounter(counter-1);

    const handleReset = () => setCounter(value);

    return(
        <>
            <h1>counterApp</h1>
            <h2 data-testid="container-counter">{ counter }</h2>
            <button onClick={handleAdd}>+1</button>
            <button onClick={handleSubstract}>-1</button>
            <button aria-label="btn-reset" onClick={handleReset}>Reset</button>
        </>
    )
}

CounterApp.prototypes = {
    value: PropTypes.number
}