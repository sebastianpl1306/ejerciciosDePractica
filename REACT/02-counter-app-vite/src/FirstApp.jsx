import PropTypes from "prop-types";

const info = {
  name: 'Sebastian',
  lastname: 'Pabon Lopez'
};

const getName = (info) =>{
  return `Hello ${info.name} ${info.lastname} welcome!`;
}

export const FirstApp = ({title, subtitle, number}) => {

  return (
    <>
      <h1>{ getName(info) }</h1>
      <h2 data-testid='test-title'>{title}</h2>
      <p>{subtitle}</p>
      <p>{subtitle}</p>
      <p>FirstApp {number+1}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  number: PropTypes.number,
  subtitle: PropTypes.string
};

FirstApp.defaultProps = {
  number: 2
}