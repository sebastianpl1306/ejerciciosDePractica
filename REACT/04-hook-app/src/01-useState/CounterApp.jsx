import { useState } from "react"

export const CounterApp = () => {
    const [state, setCounter] = useState({
      counter1: 1, 
      counter2: 2, 
      counter3: 3}
    );
    const { counter1, counter2, counter3 } = state;
  return (
    <>
      <h1>Counter1: {counter1}</h1>
      <h1>Counter2: {counter2}</h1>
      <h1>Counter3: {counter3}</h1>
      <button onClick={
        ()=>setCounter( {...state, counter1: counter1 + 1} )
      }>+1</button>
    </>
  )
}
