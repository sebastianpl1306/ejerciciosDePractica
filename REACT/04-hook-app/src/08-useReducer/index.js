export * from './TodoAdd';
export * from './TodoApp';
export * from './TodoItem';
export * from './TodoList';
export * from './todoReducer';