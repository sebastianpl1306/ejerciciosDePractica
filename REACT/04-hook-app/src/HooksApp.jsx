export const HooksApp = () => {
  return (
    <h1 className="text-primary">HooksApp</h1>
  )
}