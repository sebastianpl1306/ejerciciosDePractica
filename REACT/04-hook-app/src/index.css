body{
    padding: 20px;
}

button{
    margin-right: 10px;
}

a{
    margin: 5px;
}