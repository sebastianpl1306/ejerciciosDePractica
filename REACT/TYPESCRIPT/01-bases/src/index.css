html, body{
  background-color: #282c34;
  color: white;
  margin: 0;
  padding: 10px;
}