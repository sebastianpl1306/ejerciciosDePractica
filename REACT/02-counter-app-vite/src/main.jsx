import React from 'react'
import ReactDOM from "react-dom/client";
import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={0}></CounterApp>
        {/* <FirstApp title="Hola, soy Sebastiannn" number={123}></FirstApp> */}
    </React.StrictMode>
)