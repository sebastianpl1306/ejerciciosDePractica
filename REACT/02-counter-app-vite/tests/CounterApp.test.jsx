import { fireEvent, render, screen } from "@testing-library/react"
import { CounterApp } from "../src/CounterApp"

describe('Pruebas en el componente CounterApp', () => {
    const initial=0;

    test('Debe de hacer match con el snapshot', () => {
        const { container } = render(<CounterApp value={initial}/>);
        expect( container ).toMatchSnapshot();
    })

    test('Debe de mostrar el valor inicial de 100', () => {
        render(<CounterApp value={initial}/>);
        expect( parseInt(screen.getByTestId('container-counter').innerHTML) ).toBe(initial);
    })

    test('Debe de incrementar con el boton +1', () => {
        render(<CounterApp value={initial}/>);
        fireEvent.click( screen.getByText('+1') );

        expect( screen.getByText(`${initial + 1}`) ).toBeTruthy();
    })

    test('Debe de decrementar con el boton -1', () => {
        render(<CounterApp value={initial}/>);
        fireEvent.click( screen.getByText('-1') );

        expect( parseInt(screen.getByTestId('container-counter').innerHTML) ).toBe(initial-1);
    })

    test('Debe de funcionar el boton reset', () => {
        render(<CounterApp value={initial}/>);
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        // fireEvent.click( screen.getByText('Reset') );
        fireEvent.click( screen.getByRole('button', {name: 'btn-reset'}) );

        expect( screen.getByText( initial ) ).toBeTruthy();
    })
})