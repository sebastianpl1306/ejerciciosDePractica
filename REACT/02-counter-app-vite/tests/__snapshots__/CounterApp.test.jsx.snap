// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente CounterApp Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    counterApp
  </h1>
  <h2
    data-testid="container-counter"
  >
    0
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
